Fix HDMI preferred mode choosing

This fixes the HDMI setting to use the preferred mode instead of the
lowest resolution by default.

Upstream-Status: Pending

Signed-off-by: Jon Nettleton <jon.nettleton@gmail.com>
Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

diff --git a/EXA/src/vivante_fbdev/imx_display.c b/EXA/src/vivante_fbdev/imx_display.c
index af6dcfc..cf49003 100644
--- a/EXA/src/vivante_fbdev/imx_display.c
+++ b/EXA/src/vivante_fbdev/imx_display.c
@@ -879,6 +879,8 @@ imxDisplayGetModes(ScrnInfoPtr pScrn, const char* fbDeviceName)
 			(mode->HDisplay > 0) &&
 				(mode->VDisplay > 0)) {
 
+			mode->type = M_T_PREFERRED;
+
 			xf86PrintModeline(pScrn->scrnIndex, mode);
 			modesList = xf86ModesAdd(modesList, mode);
 		}
